### DISCUSSION TITLE
Is it possible to get the current weight of a cache?

### DISCUSSION THREAD

**Discussion by ken-schneider** [ben-manes/caffeine#822] [Q&A] [answered]
Hi, if I set up a cache with `maximumWeight`, is there an API available to calculate current weight of the cache similar to `cache.estimatedSize()`? Or is what's being returned from `cache.estimatedSize()` in a weighted cache the current weight? Thanks!

```java
Caffeine.newBuilder()
            .executor(context().dispatcher())
            .maximumWeight(someWeightInBytes)
            .weigher(new SomeWeigher())
            .buildAsync((request, executor) -> load(request));
```

**Comment by ben-manes** [comment, marked as answer]
Yes, you can use the policy api ([wiki](https://github.com/ben-manes/caffeine/wiki/Policy), [javadoc](https://www.javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/latest/com.github.benmanes.caffeine/com/github/benmanes/caffeine/cache/Policy.html)) for these configuration-specific inspections. That is a bit hidden to not pollute the main api and keep it simpler, while also not blocking users by only offering an overly generic set of methods. Since it is a bit of a dumping ground those apis are less polished, so sorry if it could be a little nicer as it was added to as needed.

```java
var eviction = cache.policy().eviction().orElseThrow()
long currentWeight = eviction.weightedSize().orElseThrow()
```
